src/bin/grin.rs

```diff
@@ -148,7 +148,7 @@ fn real_main() -> i32 {
 
 	// Initialize our global chain_type and feature flags (NRD kernel support currently).
 	// These are read via global and not read from config beyond this point.
-	global::init_global_chain_type(config.members.unwrap().server.chain_type);
+	global::init_global_chain_type(config.members.as_mut().unwrap().server.chain_type);
```

Comment:
`as_mut()` not required here?

Reply:
my understanding is that all but the last access to config.members needs to have as_mut(),
as suggested by lines 127 and 128.

Reply:
Ugh.

Reply:
Yeah - this whole init of the config needs rethinking in `grin.rs`. It looks like we have repeatedly just hacked this until it worked...

We should not need to have `mut` config here as we do not want to inadvertently write to it. But it needs to be `mut` because something else higher up in the file has modified a default via a `mut` on some _other_ piece of nested config.

:+1: on getting this working for now - but we should revisit at some point.

Reply:
i see that grin-wallet uses

```
        global::init_global_chain_type(
                config
                        .members
                        .as_ref()
                        .unwrap()
                        .wallet
                        .chain_type
                        .as_ref()
                        .unwrap()
                        .clone(),
        );
```
perhaps that's better than as_mut() ?

Reply:
do you agree with adding a global::init_global_accept_fee_base() method and associated
global::get_accept_fee_base() for the benefit of grin-wallet?
It would call global::init_global_accept_fee_base() after parsing the grin-wallet.toml file,
and then pass global::get_accept_fee_base() as 4th argument whenever calling txlib::mod::tx_fee

Reply:
I decided to have an accept_fee_base  parameter in BOTH grin-server.toml and grin-wallet.toml,
and both using global::get_accept_fee_base() to query its value. It's even possible (but not particularly useful) to set then to different values in simultaneously running server and wallet.

As a result, we can simply libtx::mod::tx_fee to no longer have an optional fee base argument. If you want to deviate from the default, you can just change it locally, like I do in some tests.